File: Ejercicios/Ejercicio3/3.java
import java.io.*;

class Persona implements Serializable 
{
    private String nombre;
    private int edad;
    private String direccion;

    public Persona() {}

    public Persona(String nombre, int edad, String direccion) 
    {
        this.nombre = nombre;
        this.edad = edad;
        this.direccion = direccion;
    }

    public void setNombre(String nombre)
     {
        this.nombre = nombre;
    }

    public void setEdad(int edad) 
    {
        this.edad = edad;
    }

    public void setDireccion(String direccion) 
    {
        this.direccion = direccion;
    }

    public String getNombre() 
    {
        return nombre;
    }

    public int getEdad() 
    {
        return edad;
    }

    public String getDireccion() 
    {
        return direccion;
    }

    public void serialize(String filename) 
    {
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(filename))) 
        {
            oos.writeObject(this);
        } catch (IOException e) 
        {
            System.err.println("Error abriendo el archivo para escritura.");
        }
    }

    public void deserialize(String filename) 
    {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(filename))) 
        {
            Persona obj = (Persona) ois.readObject();
            this.nombre = obj.nombre;
            this.edad = obj.edad;
            this.direccion = obj.direccion;
        } 
        catch (IOException | ClassNotFoundException e) 
        {
            System.err.println("Error abriendo el archivo para lectura.");
        }
    }
}

public class 3
{
    public static void main(String[] args) 
    {
        Persona p1 = new Persona("Adrian Shephard", 30, "BlacMesaEast-897");
        p1.serialize("persona.bin");

        Persona p2 = new Persona();
        p2.deserialize("persona.bin");

        System.out.println("Nombre: " + p2.getNombre());
        System.out.println("Edad: " + p2.getEdad());
        System.out.println("Direccion: " + p2.getDireccion());
    }
}

File: Ejercicios/Ejercicio3/3.py
import pickle

class Persona:
    def __init__(self, nombre="", edad=0, direccion=""):
        self.nombre = nombre
        self.edad = edad
        self.direccion = direccion

    def setNombre(self, nombre):
        self.nombre = nombre

    def setEdad(self, edad):
        self.edad = edad

    def setDireccion(self, direccion):
        self.direccion = direccion

    def getNombre(self):
        return self.nombre

    def getEdad(self):
        return self.edad

    def getDireccion(self):
        return self.direccion

    def serialize(self, filename):
        try:
            with open(filename, 'wb') as ofs:
                pickle.dump(self, ofs)
        except IOError:
            print("Error abriendo el archivo para escritura.")

    def deserialize(self, filename):
        try:
            with open(filename, 'rb') as ifs:
                obj = pickle.load(ifs)
                self.nombre = obj.nombre
                self.edad = obj.edad
                self.direccion = obj.direccion
        except IOError:
            print("Error abriendo el archivo para lectura.")


p1 = Persona("Adrian Shephard", 30, "BlacMesaEast-897")
p1.serialize("persona.bin")

p2 = Persona()
p2.deserialize("persona.bin")

print(f"Nombre: {p2.getNombre()}")
print(f"Edad: {p2.getEdad()}")
print(f"Direccion: {p2.getDireccion()}")
